### PR TITLE
408 imputation prep

### PR DESCRIPTION
**Pull request by AntonZogk**
* Convert cell_no for imputation class, plus test and test data
* Run frozen or live, plus test and test data
* Convert annual to monthly basic transformation

# Summary

Some more requirements prior to imputation
# Checklists

This pull request meets the following requirements:

- [x] installable with all dependencies recorded
- [x] runs without error
- [x] follows PEP8 and project specific conventions
- [x] appropriate use of comments, for example no descriptive comments
- [x] functions documented using Numpy style docstings
- [x] assumptions and decisions log considered and updated if appropriate
- [x] unit tests have been updated to cover essential functionality for a reasonable range of inputs and conditions
- [x] other forms of testing such as end-to-end and user-interface testing have been considered and updated as required
- [x] tests suite passes (locally as a minimum)
- [x] peer reviewed with review recorded

If you feel some of these conditions do not apply for this pull request, please
add a comment to explain why.

### CODE REVIEW

File: mbs_results/data_cleaning.py [hunk 1]
```diff
@@ -177,3 +180,96 @@ def load_manual_constructions(
...
+
+    Returns
+    -------
+    df : pd.DataFrame
+        Oringal dataframe with new_col.
+    """
+    df[new_col] = (
+        df[cell_no_col]
+        .astype(str)
+        .map(lambda x: str(5) + x[1:-1] if x[0] == str(7) else x[:-1])
```
[the remark this answers is not in the document]
Comment (Jday7879) [reply mbs_results/data_cleaning.py:271]: I prefer the current method, both take a little bit of time to understand, but the implemented method has easier to understand logic

File: mbs_results/data_cleaning.py [hunk 2]
```diff
@@ -177,3 +180,96 @@ def load_manual_constructions(
...
+        Original dataframe.
+    cell_no_col : str
+        Column name of df.
+    new_col : str
+        Column name to save the results.
+
+    Returns
+    -------
+    df : pd.DataFrame
+        Oringal dataframe with new_col.
```
Comment (Jday7879) [inline mbs_results/data_cleaning.py:266]: Spelling: Original

File: tests/test_data_cleaning.py [hunk 3]
```diff
@@ -75,3 +80,30 @@ def test_clean_and_merge():
         ["reference", "period"]
     )
     assert_frame_equal(actual_output, expected_output)
+
+
+def test_create_imputation_class():
```
Comment (Jday7879) [inline tests/test_data_cleaning.py:85]: This unit test fails on local machine, passed on Github actions so unsure about the cause of this. Failed on dypes int32 vs int64

**Review by Jday7879 — CHANGES_REQUESTED** [review]
Nice work Anton, made a few minor suggestions. Happy to merge after these changes :)

**Review by Jday7879 — CHANGES_REQUESTED** [review]
Nice work Anton! Made a few small suggestions to add litle bit more detail in before merging

File: mbs_results/data_cleaning.py [hunk 4]
```diff
@@ -177,3 +180,96 @@ def load_manual_constructions(
...
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Original dataframe.
+    target : str or list[str]
+        Column(s) to treat as non-response.
+    error_marker : str
+        Column name with error values.
+    state : str, optional
+        Function config parameter. The default is "live".
```
Comment (Jday7879) [inline mbs_results/data_cleaning.py:206]: Could you add a little more information about this parameter?

File: mbs_results/data_cleaning.py [hunk 5]
```diff
@@ -177,3 +180,96 @@ def load_manual_constructions(
...
+    error_values : list[str], optional
+        Values to ignore. The default is ['E', 'W'].
+
+    Returns
+    -------
+    Original dataframe.
+
+    """
+
+    # TODO: raise error if state is not frozen or live
```
Comment (Jday7879) [inline mbs_results/data_cleaning.py:216]: What is supposed to happen if the state is not live or frozen. Could this be added now?

File: tests/test_data_cleaning.py (hunk 3 above)
Comment (AntonZogk) [reply tests/test_data_cleaning.py:85]: Interesting one, i will double check my dev env and pandas numpy versions

Comment (AntonZogk) [reply tests/test_data_cleaning.py:85]: for reference this was developed with pandas  1.1.5   and numpy   1.19.5   python in cdh

File: mbs_results/data_cleaning.py [hunk 6]
```diff
@@ -177,3 +180,102 @@ def load_manual_constructions(
...
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Original dataframe.
+    target : str or list[str]
+        Column(s) to treat as non-response.
+    error_marker : str
+        Column name with error values.
+    state : str, optional
+        Function config parameter. The default is "live". "Live" state won't do
```
Comment (Jday7879) [inline mbs_results/data_cleaning.py:206]: Slightly pedantic but the second live shouldn't have a capital letter. It looks like "Live" and "live" are separate inputs. 

Appreciate the extra detail in the doc string though :)

**Review by Jday7879 — APPROVED** [review]
Nice work Anton. Happy for this to me merged. There is one minor and pedantic suggestion in the docstring of `run_live_or_frozen`. Happy for this to be merged before or after this change is made

File: mbs_results/data_cleaning.py (hunk 6 above)
Comment (AntonZogk) [reply mbs_results/data_cleaning.py:206]: how you spotted that ? :D , thanks Jordan will update it

**Pull request merged**